src/transformers/modeling_utils.py

```diff
@@ -2223,12 +2346,17 @@ def _find_mismatched_keys(
 
             error_msgs = []
             mismatched_keys = []
+            offload_index = {} if device_map is not None and "disk" in device_map.values() else None
+            if offload_state_dict:
+                state_dict_folder = tempfile.mkdtemp()
+                state_dict_index = {}
+            else:
+                state_dict_folder = None
+                state_dict_index = None
```

Comment:
For offloaded weights (either on disk or temp offload of the CPU weights) this index contains the map param_name -> metadata (shape and dtype)